openlibrary/core/stats.py

```diff
@@ -41,19 +41,19 @@ def put(key, value):
     "Records this ``value`` with the given ``key``. It is stored as a millisecond count"
     global client
     if client:
-        l.debug("Putting %s as %s"%(value, key))
+        l.debug("Putting %s as %s" % (value, key))
         client.timing(key, value)
 
-def increment(key, n=1):
+def increment(key, n=1, rate=1.0):
```

Comment:
1.0 or 0.1 ?  1/10th of a second

Reply:
https://github.com/internetarchive/openlibrary/issues/3782#issuecomment-690614459

https://statsd.readthedocs.io/en/v3.2.1/reference.html#incr

rate: a sample rate, a float between 0 and 1. Will only send data this percentage of the time. The statsd server will take the sample rate into account for counters.